File: BussinesHub.Front.v2/src/models/Company.ts
import { Store } from "./Store";
import { Analitic } from "./Analitic";

export interface Company {
  id: number;
  name: string;
  identificationNumber: string;
  adress: string;
  city: string;
  postalCode: string;
  country: string;
  stores: Store[];
  analitics: Analitic[];
}

export class CompanyClass implements Company {
  id = 0;
  name = "";
  identificationNumber = "";
  adress = "";
  city = "";
  postalCode = "";
  country = "";
  stores = [];
  analitics = [];
}

File: BussinesHub.API/BH.Model/General/Company.cs
namespace BH.Model.General
{
	public class Company
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public string IdentificationNumber { get; set; }
		public string Adress { get; set; }
		public string City { get; set; }
		public string PostalCode { get; set; }
		public string Country { get; set; }
		public string PhoneNumber { get; set; }
		public virtual ICollection<Store> Stores { get; set; } = new List<Store>();
		public virtual ICollection<Analitic>? Analitics { get; set; } = new List<Analitic>();
		public virtual ICollection<User> Users { get; set; } = new List<User>();
		public virtual ICollection<Product> Products { get; set; } = new List<Product>();
		public virtual ICollection<Category> Categories { get; set; } = new List<Category>();
	}
}